File: examples/greeting-service/src/main/java/greeting/example/Greeting.java
package greeting.example;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import io.micronaut.core.annotation.Introspected;

@Introspected
public class Greeting {

    private final long id;
    private final String content;

    @JsonCreator
    public Greeting(@JsonProperty("id") long id, @JsonProperty("content") String content) {
        this.id = id;
        this.content = content;
    }

    public long getId() {
        return id;
    }

    public String getContent() {
        return content;
    }
}

File: examples/greeting-service-kotlin/src/main/kotlin/greeting/example/Greeting.kt
package greeting.example

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Introspected

@Introspected
class Greeting @JsonCreator constructor(
        @JsonProperty("id") val id: Long,
        @JsonProperty("content") val content: String) {

}